regions/core/pixcoord.py

```diff
@@ -1,26 +1,152 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 from __future__ import absolute_import, division, print_function, unicode_literals
+import numbers
+import numpy as np
+from astropy.coordinates import SkyCoord
 
 __all__ = ['PixCoord']
 
 
 class PixCoord(object):
+    """Pixel coordinates.
+
+    This class can represent scalar or array pixel coordinates.
+
+    TODO: see examples below and / or in high-level docs section.
+
+    The data members are either numbers or Numpy arrays
+    (not `~astropy.units.Quantity` objects with unit "pixel").
+
+    Given a `astropy.wcs.WCS` object, it can be transformed to and from a
+    `~astropy.coordinates.SkyCoord` object.
+
+    TODO: 2-dim or n-dim arrays of pixel coordinates are not supported yet.
+    Having those would be very useful, e.g. for arrays of pixel coordinates in
+    a 2-dim image.
+
+    Parameters
+    ----------
+    x : float or array-like
+        Pixel coordinate x value
+    y : float or array-like
+        Pixel coordinate y value
+    origin : int
+        Origin of pixel coordinates (usually 0 or 1)
+    mode : {'all', 'wcs'}
+        Whether to do the transformation including distortions ('all')
+        or only including only the core WCS transformation ('wcs').
+
+    Examples
+    --------
+    >>> from regions import PixCoord
+
+    Scalar pixel coordinate:
+
+    >>> pixcoord = PixCoord(x=1, y=11)
+    >>> print(pixcoord)
+    PixCoord
+    x : 1
+    y : 11
+
+    Array pixel coordinates:
+
+    >>> pixcoord = PixCoord(x=[1, 2], y=[11, 12])
+    >>> print(pixcoord)
+    PixCoord
+    x : [1 2]
+    y : [11 12]
     """
-    Class representing a collection of pixel coordinates
-    """
+    # We define defaults origin and mode here for WCS transforms.
+    # These defaults are used in a few places as default keyword arguments to methods.
+    # The purpose is to ensure consistency across the codebase.
+    # They should not be modified by users, it's an implementation detail.
+    # TODO: is this the best place to put those defaults?
+    # Should we put them somewhere else?
+    # Or remove them and try to achieve consistency manually?
+    _DEFAULT_ORIGIN = 0
+    _DEFAULT_MODE = 'all'
```

Comment:
> I personally would say that instead of having these as module/class level variables, they can be options that are specified during transformations.

I agree. Unless others speak up in favour of keeping `origin` and / or `mode` as data members on the `PixCoord` class, I'll make this change.